a10_octavia/controller/worker/tasks/a10_network_tasks.py

```diff
@@ -658,3 +660,30 @@ def revert(self, result, loadbalancer, amp_data=None, update_dict=None,
         except Exception as e:
             LOG.error('Failed to remove QoS policy: %s from port: %s due '
                       'to error: %s', orig_qos_id, amp_data.vrrp_port_id, e)
+
+
+class GetSubnetVLANIDParent(object):
+    """Get the Subnet VLAN_ID"""
+
+    def get_vlan_id(self, subnet_id):
+        network_id = self.network_driver.get_subnet(subnet_id).network_id
+        network = self.network_driver.get_network(network_id)
+        if network.provider_network_type != 'vlan':
+            raise
+        return network.provider_segmentation_id
+
+
+class GetVipSubnetVLANID(GetSubnetVLANIDParent, BaseNetworkTask):
```

Comment:
We don't do getters in taskflow unless it's an edge case where getting it upfront is the only way to use the decider. All of these should be moved to the tasks that are using them.

Also it's bad design in general to do pass through functions such as this.